fil_fungible_token/src/token/mod.rs

```diff
@@ -294,6 +296,28 @@ where
         Ok(())
     }
 
+    /// Sets the allowance to a specified amount, returning the old allowance
+    pub fn set_allowance(
+        &mut self,
+        owner: &Address,
+        operator: &Address,
+        amount: &TokenAmount,
+    ) -> Result<TokenAmount> {
+        // Handle special revoke allowance case
+        if amount.is_zero() {
+            let old_allowance = self.allowance(owner, operator)?;
+            self.revoke_allowance(owner, operator)?;
```

Comment:
Ok, please document the intent of handling this special case, which is to avoid initialising addresses unnecessarily.

I was confused earlier here and thought this was calling through to a state method, rather than self method. But since we're optimising, change `revoke_allowance` to return the old allowance (via a similar change to the state method). It's already being read and discarded at `Map::delete`.